app/src/main/java/com/example/joshuarodriguez/mvpcalculator/mvp/mvp/view/CalculatorView.java

```diff
@@ -237,7 +235,17 @@ public void clearCalculatorInputField() {
         mEditTextInputField.setText(EMPTY_TEXT);
     }
 
-    private void hideKeyboard() {
+    public void hideKeyboard() {
         mEditTextInputField.setInputType(InputType.TYPE_NULL);
     }
+
+    private String getStringResource(@StringRes int stringResourceId) {
+        AppCompatActivity appCompatActivity = getActivity();
+        return (appCompatActivity != null) ? appCompatActivity.getString(stringResourceId) : "";
```

Comment:
You should not use this type of method. Because if appCompatActivity is null you dont need continue with that particular execution. For example 
`
AppCompatActivity appCompatActivity = getActivity();
if(appCompatActivity==null){
return;
}
........................
`